packages/taquito-sapling/src/sapling-transaction-viewer.ts

```diff
@@ -1,23 +1,275 @@
-import * as sapling from '@airgap/sapling-wasm'
+import * as sapling from '@airgap/sapling-wasm';
+import BigNumber from 'bignumber.js';
+import { hex2buf, mergebuf } from '@taquito/utils';
+import { CommitmentsAndCiphertexts, RpcClient, RpcClientInterface } from '@taquito/rpc';
+import blake from 'blakejs';
+import { openSecretBox } from '@stablelib/nacl';
+import { InMemoryViewingKey } from './in-memory-viewing-key';
+import { bufToUint8Array, convertValueToBigNumber, readableFormat } from './helpers';
 
+const KDF_KEY = 'KDFSaplingForTezosV1';
+const OCK_KEY = 'OCK_keystringderivation_TEZOS';
+
+export interface SaplingIncomingAndOutgoingTransaction {
+  incoming: SaplingIncomingTransaction[];
+  outgoing: SaplingOutgoingTransaction[];
+}
+export interface SaplingIncomingTransaction {
+  value: BigNumber;
+  memo: string;
+  paymentAddress: string;
+  isSpent: boolean;
+}
+
+export interface SaplingOutgoingTransaction {
+  value: BigNumber;
+  memo: string;
+  paymentAddress: string;
+}
+
+export interface SaplingTransactionPropertiesRaw {
+  value: Uint8Array;
+  memo: Uint8Array;
+  paymentAddress: Uint8Array;
+  rcm: Uint8Array;
+}
+/**
+ * @description Allows to retrieve and decrypt sapling transactions
+ */
 export class SaplingTransactionViewer {
-    #viewingKey: Buffer;
-    constructor(viewingKey: Buffer) {
-        this.#viewingKey = viewingKey
+  #viewingKeyProvider: InMemoryViewingKey;
```

Comment:
what's the reasoning behind having a  # symbol in the beginning?

Reply:
It means private. This was a recommendation from https://github.com/ecadlabs/taquito/issues/1184 that I started to apply.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

Reply:
ah I see, is this also a replacement to the `private` keyword??

Reply:
yes